include/seqan3/utility/tuple/common_pair.hpp

```diff
@@ -0,0 +1,743 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::common_pair.
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <utility>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3
+{
+
+/*!\brief A [std::pair](https://en.cppreference.com/w/cpp/utility/pair) implementation that incorporates most changes
+ *        from C++23's standard library.
+ * \ingroup utility_tuple
+ */
+template <class T1, class T2>
+struct common_pair : public std::pair<T1, T2>
+{
+private:
+    //!\brief The underlying std::pair type.
+    using pair_t = std::pair<T1, T2>;
```

Comment:
I find `std_pair_t` a bit more clunky, and you would have to know that it inherits from std::pair.
An alternative would be our beloved `base_t`.

Reply:
hm I'd prefer base_t than because it at least tells you that it's the type we inherit from. pair_t doesn't really tell me much and might be confusing because you are actually defining a "pair type".